### PR TITLE
Allows the patreon list to be used as a whitelist

### PR DESCRIPTION
**Pull request by francinum**
## Changelog
:cl:
config: the Patreon donators list can now be used as a whitelist.
/:cl:

### CODE REVIEW

File: code/game/world.dm [hunk 1]
```diff
@@ -45,7 +45,10 @@ GLOBAL_VAR(restart_counter)
 		world.log = file("[GLOB.log_directory]/dd.log") //not all runtimes trigger world/Error, so this is the only way to ensure we can see all of them.
 #endif
 	if(CONFIG_GET(flag/usewhitelist))
-		load_whitelist()
+		if(CONFIG_GET(flag/use_patrons_as_whitelist))
+			GLOB.whitelist = GLOB.patrons
+		else
+			load_whitelist()
```
Comment (Archanial) [inline code/game/world.dm]: I feel like you should load both, the actual whitelist and the patron whitelist

File: config/config.txt [hunk 2]
```diff
@@ -248,6 +248,11 @@ CHECK_RANDOMIZER
 ## Set to prevent anyone but those ckeys listed in config/whitelist.txt and config/admins.txt from joining your server
 #USEWHITELIST
 
+## Use the patron list as a whitelist. 'USEWHITELIST' must also be enabled.
```
Comment (francinum) [inline config/config.txt]:
```suggestion
## Use the patron list as a whitelist. 'USEWHITELIST' must also be enabled. The two lists will be joined together.
```

**Review by Archanial — DISMISSED** [review]

**Comment by RDS88-beeman** [comment]
this feels like a solid planndemic moment. Are patreons gonna get their own private server?

**Review by Crossedfall — DISMISSED** [review]

File: config/config.txt [hunk 3]
```diff
@@ -248,6 +248,11 @@ CHECK_RANDOMIZER
 ## Set to prevent anyone but those ckeys listed in config/whitelist.txt and config/admins.txt from joining your server
 #USEWHITELIST
 
+## Use the patron list as a whitelist. 'USEWHITELIST' must also be enabled. The two lists will be joined together.
+#USE_PATRONS_AS_WHITELIST
+
+
```
Comment (Crossedfall) [inline config/config.txt]:
```suggestion
```

**Review by Crossedfall — APPROVED** [review]

**Comment by Tsar-Salat** [comment]
Donator Gaming

**Pull request merged**